ludwig/schema/common_fields.py

```diff
@@ -0,0 +1,124 @@
+from dataclasses import Field
+from typing import Optional
+
+from ludwig.schema import utils as schema_utils
+from ludwig.schema.metadata import COMMON_METADATA
+from ludwig.schema.metadata.parameter_metadata import ParameterMetadata
+from ludwig.utils.torch_utils import initializer_registry
+
+
+def DropoutField(default: float = 0.0, description: str = None, parameter_metadata: ParameterMetadata = None) -> Field:
+    description = description or (
+        "Default dropout rate applied to fully connected layers. "
+        "Increasing dropout is a common form of regularization to combat overfitting. "
+        "The dropout is expressed as the probability of an element to be zeroed out (0.0 means no dropout)."
+    )
+    parameter_metadata = parameter_metadata or COMMON_METADATA["dropout"]
+    return schema_utils.FloatRange(
+        default=default,
+        min=0,
+        max=1,
+        description=description,
+        parameter_metadata=parameter_metadata,
+    )
+
+
+def ResidualField(
+    default: bool = False, description: str = None, parameter_metadata: ParameterMetadata = None
+) -> Field:
+    description = description or (
+        "Whether to add a residual connection to each fully connected layer block. "
+        "Requires all fully connected layers to have the same `output_size`."
+    )
+    parameter_metadata = parameter_metadata or COMMON_METADATA["residual"]
+    return schema_utils.Boolean(
```

Comment:
idk if this is that much more readable/efficient but since we're passing all local variables you could say e.g. `Boolean(**locals())`?

Reply:
That's true, but a little dangerous if someone ever adds a new local var.